hadoop-ozone/recon/src/main/java/org/apache/hadoop/ozone/recon/scm/ReconNodeManager.java

```diff
@@ -276,8 +287,15 @@ public RegisteredCommand register(
             datanodeDetails.getUuid());
       }
     }
-    return super.register(datanodeDetails, nodeReport, pipelineReportsProto,
-        layoutInfo);
+    try {
+      return super.register(datanodeDetails, nodeReport, pipelineReportsProto,
+          layoutInfo);
+    } catch (InvalidTopologyException invalidTopologyException) {
+      LOG.error("InvalidTopologyException error occurred : {}", invalidTopologyException.getMessage());
+      reconContext.updateHealthStatus(false);
+      reconContext.getErrors().add(ReconContext.ErrorCode.INVALID_NETWORK_TOPOLOGY);
+    }
+    return null;
```

Comment:
Do we need return RegisteredCommand instead of null? with status failure

Reply:
As discussed, Recon will continue to return null as Error codes for RegisteredCommand are not compatible between SCM and Recon.